docs/interfaces.md

```diff
@@ -0,0 +1,431 @@
+        X509_STORE* root_certs_apps, // root cert store for application orgs
+
+        // tlcc session
+        session_keys_t session_keys,
+};
+
+typedef struct kvs_t {};
+
+typedef struct session_keys_t {};
+```
+
+
+# ECC
+
+## Interface:
+The ECC interface is implemented through a "normal" chaincode interface. Methods with leading underscores
+are treated as FPC commands. Normal `invoke` invocations are forwarded to a FPC chaincode enclave.
+
+```go
+// chaincode interface (exposed to admin/clients) implemented by invoke
+func Invoke(stub shim.ChaincodeStubInterface) pb.Response {}
+
+// triggered by an admin
+CreateEnclave(params CC_Parameters) (Credentials, error)
+
+// key generatation 
+GenerateCCKeys(params CC_Parameters) (CCKey_registration_message, error)
+ExportCCKeys(credentals Credentials) (ExportMessage, error)
+ImportCCKeys(params CC_Parameters) (CCKey_registration_message, error)
+
+// chaincode invoke
+ChaincodeInvoke(transactionProposal []byte, ctx interface{}) (response, error)
+```
+
+```go
+// ECC must trigger this when sealedState is available using load() and ecc starts; this is not really a public interface
+createEnclaveWithSealedState(sealedState []byte) error
+
+// Storage interface
+// TODO how to store sealedState, where and under what identifier (e.g. chaincode_id);
+load(chaincode_id string) (sealedState []byte, err error)
+store(chaincode_id string, sealedState []byte) error
+```
+
+This interface is implemented by ECC to let a chaincode enclave call into the peer
+
+```c++
+
+// logging
+void ocall_print_string([in, string] const char *str);
+
+
+// state access
+// TBD does this key already contain the chaincode_id/namespace?
+void ocall_get_state(
+        [in, string] const char *key, 
+        [out, size=max_val_len] uint8_t *val, uint32_t max_val_len, [out] uint32_t *val_len,
+        [in, out] sgx_cmac_128bit_tag_t *cmac,
+        [user_check] void *u_shim_ctx);
+
+void ocall_put_state(
+        [in, string] const char *key,
+        [in, size=val_len] uint8_t *val, uint32_t val_len,
+        [user_check] void *u_shim_ctx);
+
+void ocall_get_state_by_partial_composite_key(
+        [in, string] const char *comp_key,
+        [out, size=max_len] uint8_t *values, uint32_t max_len, [out] uint32_t *values_len,
+        [in, out] sgx_cmac_128bit_tag_t *cmac,
+        [user_check] void *u_shim_ctx);
+
+
+// verify msp with the help of tlcc
+void ocall_verify_msp_id(
+        [in, string] const char *msp_id,
+        [in, size=32] uint8_t *nonce,
+        [out] bool isValid, 
+        [out] sgx_cmac_128bit_tag_t *cmac);
+
+
+// attestation
+void ocall_get_spid(); // TBD: is spid only needed for EPID?
+
+// creates an attestation evidence; the evidence_request includes all the form of evidence (EPID/DCAP), the quote, and additional information required to complete the evidence request
+void ocall_create_evidence(
+        [in] evidence_request
+        [out] evidence
+);
```

Comment:
As mentioned above, i think we should delegate these ocalls/ecalls to the implementation of  #412 and just rely on the (local call) API (to be) defined by #412.

To some extent you already assume the existance of such a module by the `attestation.EvidenceService` in your state?

Reply:
added reference to #412

Reply:
i removed the SPID ocall but kept the ocall_create_evidence with a note that more details are in #412 and yes therefore we also should keep the reference of attestation.EvidenceService